main/src/main/java/org/mobilitydata/gtfsvalidator/validator/ShapeIncreasingDistanceValidator.java

```diff
@@ -47,57 +56,119 @@ public void validate(NoticeContainer noticeContainer) {
       for (int i = 1; i < shapeList.size(); ++i) {
         GtfsShape prev = shapeList.get(i - 1);
         GtfsShape curr = shapeList.get(i);
-        if (prev.hasShapeDistTraveled()
-            && curr.hasShapeDistTraveled()
-            && prev.shapeDistTraveled() >= curr.shapeDistTraveled()) {
+        if (!(prev.hasShapeDistTraveled() || curr.hasShapeDistTraveled())) {
+          continue;
+        }
+        if (prev.shapeDistTraveled() > curr.shapeDistTraveled()) {
+          noticeContainer.addValidationNotice(new DecreasingShapeDistanceNotice(prev, curr));
+          continue;
+        }
+        if (prev.shapeDistTraveled() != curr.shapeDistTraveled()) {
+          continue;
+        }
+        // equal shape_dist_traveled and different coordinates
+        if (!(curr.shapePtLon() == prev.shapePtLon() && curr.shapePtLat() == prev.shapePtLat())) {
           noticeContainer.addValidationNotice(
-              new DecreasingOrEqualShapeDistanceNotice(
-                  curr.shapeId(),
-                  curr.csvRowNumber(),
-                  curr.shapeDistTraveled(),
-                  curr.shapePtSequence(),
-                  prev.csvRowNumber(),
-                  prev.shapeDistTraveled(),
-                  prev.shapePtSequence()));
+              new EqualShapeDistanceDiffCoordinatesNotice(prev, curr));
+        } else {
+          // equal shape_dist_traveled and same coordinates
+          noticeContainer.addValidationNotice(
+              new EqualShapeDistanceSameCoordinatesNotice(prev, curr));
         }
       }
     }
   }
 
   /**
-   * When sorted on `shapes.shape_pt_sequence` key, shape points should have strictly increasing
+   * When sorted on {@code shapes.shape_pt_sequence} key, shape points must have strictly increasing
```

Comment:
Same as described in RULES.md, I would go with "must not have decreasing values" since the condition verified is `prev.shapeDistTraveled() > curr.shapeDistTraveled()`.
```suggestion
   * When sorted on {@code shapes.shape_pt_sequence} key, shape points must not have strictly decreasing
```